base_comment_template/models/base_comment_template.py

```diff
@@ -85,3 +117,19 @@ def name_get(self):
                 name += " (%s)" % ", ".join(item.model_ids.mapped("name"))
             res.append((item.id, name))
         return res
+
+    def search(self, args, offset=0, limit=None, order=None, count=False):
```

Comment:
This is not the idea, but to add the `search` attribute in the field `model_ids`, and put the domain in the other model `[("model_ids", "=", self._name)]`.

Reply:
Ok, I think I understand you (and I understood) but that is not possible because a "basic" user does not have access to  `ir.model` model and get an ACL error causing it not to reach the `search` method.

Reply:
But it doesn't reach the model, it reaches the `search` method of your field, and then you do there the `sudo` operation.